app/services/users.js

```diff
@@ -0,0 +1,47 @@
+const { User: UserModel } = require('../models');
+const { databaseError, badRequestError } = require('../errors');
+const logger = require('../logger');
+const config = require('../../config').common.business;
+const { generateHash } = require('../helpers/crypt_texts');
+
+const pathLogger = 'services:users';
+
+exports.getUserByEmail = async email => {
+  try {
+    const result = await UserModel.findOne({ where: { email } });
+    return result;
+  } catch (error) {
+    logger.error(`${pathLogger} --- Error db getting user --- ${error}`);
+    throw databaseError('Error getting user');
+  }
+};
+
+const emailIsValid = email => {
+  const regex = new RegExp(config.allowedEmailDomains);
+  return regex.test(email);
+};
+
+const passwordIsValid = password => /^([a-zA-Z0-9_-]){8}$/.test(password);
+
+exports.createUser = async ({ name, lastName, email, password }) => {
+  if (!emailIsValid(email)) {
+    throw badRequestError('Allowed domain for email are domains of Wolox');
+  }
+
+  if (!passwordIsValid(password)) {
+    throw badRequestError('The password does not meet the requirements');
+  }
+
+  try {
+    const creationResult = await UserModel.create({
+      name,
+      lastName,
+      password: generateHash(password),
```

Comment:
This is fine, but maybe hash the password in the controller, it is good to keep the services as logic free as possible